src/core/main.qml

```diff
@@ -188,14 +219,18 @@ Window {
                             pageView.currentItem.focus = true
                             checkWordset()
                         }
-            );
+                        );
         }
         else {
             // Register voices-resources for current locale, updates/downloads only if
             // not prohibited by the settings
             if (!DownloadManager.areVoicesRegistered()) {
                 if (DownloadManager.updateResource(
                             DownloadManager.getVoicesResourceForLocale(ApplicationSettings.locale))) {
+                    DownloadManager.downloadFinished.connect(function() {
+                        checkWordset();
+                        DownloadManager.downloadFinished.disconnect(arguments.callee);
+                    });
```

Comment:
This connect() it not necessary, because we trigger checkWordset() on DownloadManager.resourceRegistered() one line below. Only one of the two is necessary